### PR TITLE
Refactor to use common library from runtime operator

### LINKED ISSUE #235: Refactor to use the common library from Runtime Component Operator
The Runtime Component Operator v0.4.0 has the base/common library. Refactor Appsody Operator to use that. It'll be similar to how Open Liberty Operator used Appsody Operator. 

Note that the `BaseApplication` was renamed to `BaseComponent`.

### PR DESCRIPTION
**Pull request by edavidj**
**What this PR does / why we need it?**:

- Removes `pkg/utils` and `pkg/common` from appsody operator. May need to recreate the utils package down the line for appsody specific utilities, similar to how OL operator functions.
- Includes changes directly to the runtime component CRD and controller, necessary to build appsody with that repos utilities/common libraries. (Sidecar containers, cert secret ref, ApplicationName, etc.)
- If there are additional changes that I missed and should be picking up in this PR let me know

**Does this PR introduce a user-facing change?**

- [ ] User guide
- [ ] `CHANGELOG.md`

**Which issue(s) this PR fixes**:

Fixes #235

### CODE REVIEW

**Comment by leochr** [comment]
@edavidj Thanks for the PR. Looks good. Some additional changes made to RC operator:
- sidecarContainer description was revised in RC operator's user guide
- additional content about shortname 

@navidsh @arturdzm @JusteenR Please review to ensure the functionalities you delivered to RC operator are reflected correctly. Thank you.

File: pkg/controller/appsodyapplication/appsodyapplication_controller.go [hunk 1]
```diff
@@ -529,10 +549,10 @@ func (r *ReconcileAppsodyApplication) Reconcile(request reconcile.Request) (reco
 		if isKnativeSupported {
 			ksvc := &servingv1alpha1.Service{ObjectMeta: defaultMeta}
 			err = r.CreateOrUpdate(ksvc, instance, func() error {
-				appsodyutils.CustomizeKnativeService(ksvc, instance)
+				oputils.CustomizeKnativeService(ksvc, instance)
 				if r.IsOpenShift() {
-					ksvc.Annotations = appsodyutils.MergeMaps(ksvc.Annotations, appsodyutils.GetOpenShiftAnnotations(instance))
-					ksvc.Spec.Template.ObjectMeta.Annotations = appsodyutils.MergeMaps(appsodyutils.GetConnectToAnnotation(instance), ksvc.Spec.Template.ObjectMeta.Annotations)
+					ksvc.Annotations = oputils.MergeMaps(ksvc.Annotations, autils.GetOpenShiftAnnotations(instance))
```
Comment (navidsh) [inline pkg/controller/appsodyapplication/appsodyapplication_controller.go]: I guess at some point we'd need to add the `GetOpenShiftAnnotations` into Runtime Component operator. We can do that in a separate PR. Once that one is merged, we can probably get rid of the Appsody utils

Comment (edavidj) [reply pkg/controller/appsodyapplication/appsodyapplication_controller.go]: Yeah I was hoping to aim for that. I'll remove the utils in a separate PR when that's resolved in Runtime Component operator. Although it may be worth adding it back down the line if more appsody specific functionality comes along, similar to OL operator utils.

File: pkg/apis/appsody/v1beta1/zz_generated.openapi.go [hunk 2]
```diff
@@ -11,15 +11,15 @@ import (
...
-		"github.com/appsody/appsody-operator/pkg/apis/appsody/v1beta1.AppsodyApplicationSpec":        schema_pkg_apis_appsody_v1beta1_AppsodyApplicationSpec(ref),
-		"github.com/appsody/appsody-operator/pkg/apis/appsody/v1beta1.AppsodyApplicationStatus":      schema_pkg_apis_appsody_v1beta1_AppsodyApplicationStatus(ref),
-		"github.com/appsody/appsody-operator/pkg/apis/appsody/v1beta1.AppsodyRoute":                  schema_pkg_apis_appsody_v1beta1_AppsodyRoute(ref),
-		"github.com/appsody/appsody-operator/pkg/apis/appsody/v1beta1.ServiceBindingConsumes":        schema_pkg_apis_appsody_v1beta1_ServiceBindingConsumes(ref),
-		"github.com/appsody/appsody-operator/pkg/apis/appsody/v1beta1.ServiceBindingProvides":        schema_pkg_apis_appsody_v1beta1_ServiceBindingProvides(ref),
-		"github.com/appsody/appsody-operator/pkg/apis/appsody/v1beta1.StatusCondition":               schema_pkg_apis_appsody_v1beta1_StatusCondition(ref),
+		"github.com/appsody-operator/pkg/apis/appsody/v1beta1.AppsodyApplication":            schema_pkg_apis_appsody_v1beta1_AppsodyApplication(ref),
```
[the remark this answers is not in the document]
Comment (edavidj) [reply pkg/apis/appsody/v1beta1/zz_generated.openapi.go]: Oh I think that's a result of putting the project in the wrong part of my gopath. It's frustrating that operator-sdk relies on that when generating openapi files rather than the module name itself :( . I'll fix that in the next commit.

File: pkg/controller/appsodyapplication/appsodyapplication_controller.go (hunk 1 above)
Comment (navidsh) [reply pkg/controller/appsodyapplication/appsodyapplication_controller.go]: Make sense, thx!

File: pkg/apis/appsody/v1beta1/zz_generated.openapi.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (navidsh) [reply pkg/apis/appsody/v1beta1/zz_generated.openapi.go]: Yea, it's a bit annoying

File: doc/user-guide.md [hunk 3]
```diff
@@ -55,8 +55,10 @@ Each `AppsodyApplication` CR must at least specify the `applicationImage` parame
 | `pullPolicy` | The policy used when pulling the image.  One of: `Always`, `Never`, and `IfNotPresent`. |
 | `pullSecret` | If using a registry that requires authentication, the name of the secret containing credentials. |
 | `initContainers` | The list of [Init Container](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.14/#container-v1-core) definitions. |
+| `sidecarContainers` | The list of `sidecar` containers. These are additional containers to be added to the pods. Note: Sidecar containers should not be name `app`. |
 | `architecture` | An array of architectures to be considered for deployment. Their position in the array indicates preference. |
 | `service.port` | The port exposed by the container, directing traffic to the application. | 
+| `service.portName` | The name for the port exposed by the container. |
```
[the remark this answers is not in the document]
Comment (navidsh) [reply doc/user-guide.md]: Thx for adding `service. certificateSecretRef`. Please add `route.annotations` too

File: deploy/crds/appsody.dev_appsodyapplications_crd.yaml [hunk 4]
```diff
@@ -68,6 +68,8 @@ spec:
           properties:
             applicationImage:
               type: string
+            applicationName:
```
Comment (navidsh) [inline deploy/crds/appsody.dev_appsodyapplications_crd.yaml]: Please add:
- this file to the `release/daily` directory
- also don't forget to update Role/ClusterRole and add to the `release/daily`

File: deploy/role.yaml [hunk 5]
```diff
@@ -15,6 +15,7 @@ rules:
   - configmaps
   - secrets
   - serviceaccounts
+  - namespaces
```
Comment (leochr) [inline deploy/role.yaml]: @navidsh is namespaces needed for Role?

Comment (navidsh) [reply deploy/role.yaml]: Removed. Thx
